### PR TITLE
display subject when we have sync issues like network errors

### PR DESCRIPTION
**Pull request by mgallien**

### CODE REVIEW

**Comment by mgallien** [comment]
before
image

after
image

**Comment by mgallien** [comment]
@nimishavijay @jancborchardt @karlitschek 
any feedback on the exact message ?
we want to tell the user that the error is a network error
I may be able to fix the missing folder information but it may be hard to tell in a precise/reliable way why there is a network issue

**Comment by mgallien** [comment]
new version
image

File: src/libsync/bulkpropagatorjob.cpp [hunk 1]
```diff
@@ -341,10 +341,11 @@ void BulkPropagatorJob::slotStartUpload(SyncFileItemPtr item,
 
 void BulkPropagatorJob::slotOnErrorStartFolderUnlock(SyncFileItemPtr item,
                                                      SyncFileItem::Status status,
-                                                     const QString &errorString)
+                                                     const QString &errorString,
+                                                     ErrorCategory errorCategory)
```
Comment (claucambra) [inline src/libsync/bulkpropagatorjob.cpp:345]:
```suggestion
                                                     const ErrorCategory errorCategory)
```

File: src/libsync/bulkpropagatorjob.h [hunk 2]
```diff
@@ -82,7 +82,8 @@ private slots:
     // invoked on internal error to unlock a folder and faile
     void slotOnErrorStartFolderUnlock(OCC::SyncFileItemPtr item,
                                       OCC::SyncFileItem::Status status,
-                                      const QString &errorString);
+                                      const QString &errorString,
+                                      OCC::ErrorCategory errorCategory);
```
Comment (claucambra) [inline src/libsync/bulkpropagatorjob.h:86]:
```suggestion
                                      const OCC::ErrorCategory errorCategory);
```

File: src/libsync/bulkpropagatorjob.h [hunk 3]
```diff
@@ -108,7 +109,8 @@ private slots:
 
     void done(SyncFileItemPtr item,
               SyncFileItem::Status status,
-              const QString &errorString);
+              const QString &errorString,
+              ErrorCategory category);
```
Comment (claucambra) [inline src/libsync/bulkpropagatorjob.h:113]:
```suggestion
              const ErrorCategory category);
```

File: src/libsync/discoveryphase.h [hunk 4]
```diff
@@ -307,7 +309,7 @@ class DiscoveryPhase : public QObject
     QStringList _listExclusiveFiles;
 
 signals:
-    void fatalError(const QString &errorString);
+    void fatalError(const QString &errorString, OCC::ErrorCategory errorCategory);
```
Comment (claucambra) [inline src/libsync/discoveryphase.h:312]:
```suggestion
    void fatalError(const QString &errorString, const OCC::ErrorCategory errorCategory);
```

File: src/libsync/discoveryphase.h [hunk 5]
```diff
@@ -320,7 +322,7 @@ class DiscoveryPhase : public QObject
       */
     void silentlyExcluded(const QString &folderPath);
 
-    void addErrorToGui(SyncFileItem::Status status, const QString &errorMessage, const QString &subject);
+    void addErrorToGui(SyncFileItem::Status status, const QString &errorMessage, const QString &subject, OCC::ErrorCategory category);
```
Comment (claucambra) [inline src/libsync/discoveryphase.h:325]:
```suggestion
    void addErrorToGui(SyncFileItem::Status status, const QString &errorMessage, const QString &subject, const OCC::ErrorCategory category);
```

File: src/libsync/owncloudpropagator.cpp [hunk 6]
```diff
@@ -221,7 +221,7 @@ void blacklistUpdate(SyncJournalDb *journal, SyncFileItem &item)
     }
 }
 
-void PropagateItemJob::done(SyncFileItem::Status statusArg, const QString &errorString)
+void PropagateItemJob::done(SyncFileItem::Status statusArg, const QString &errorString, ErrorCategory category)
```
Comment (claucambra) [inline src/libsync/owncloudpropagator.cpp:224]:
```suggestion
void PropagateItemJob::done(SyncFileItem::Status statusArg, const QString &errorString, const ErrorCategory category)
```

File: src/libsync/owncloudpropagator.h [hunk 7]
```diff
@@ -721,7 +725,7 @@ class CleanupPollsJob : public QObject
     void start();
 signals:
     void finished();
-    void aborted(const QString &error);
+    void aborted(const QString &error, ErrorCategory errorCategory);
```
Comment (claucambra) [inline src/libsync/owncloudpropagator.h:728]:
```suggestion
    void aborted(const QString &error, const ErrorCategory errorCategory);
```

File: src/libsync/progressdispatcher.h [hunk 8]
```diff
@@ -283,7 +288,7 @@ class OWNCLOUDSYNC_EXPORT ProgressDispatcher : public QObject
     /**
      * @brief: the item was completed by a job
      */
-    void itemCompleted(const QString &folder, const OCC::SyncFileItemPtr &item);
+    void itemCompleted(const QString &folder, const OCC::SyncFileItemPtr &item, OCC::ErrorCategory category);
```
Comment (claucambra) [inline src/libsync/progressdispatcher.h:291]:
```suggestion
    void itemCompleted(const QString &folder, const OCC::SyncFileItemPtr &item, const OCC::ErrorCategory category);
```

File: src/libsync/propagateupload.cpp [hunk 9]
```diff
@@ -629,10 +629,10 @@ void PropagateUploadFileCommon::slotPollFinished()
     finalize();
 }
 
-void PropagateUploadFileCommon::done(SyncFileItem::Status status, const QString &errorString)
+void PropagateUploadFileCommon::done(SyncFileItem::Status status, const QString &errorString, ErrorCategory category)
```
Comment (claucambra) [inline src/libsync/propagateupload.cpp:632]:
```suggestion
void PropagateUploadFileCommon::done(SyncFileItem::Status status, const QString &errorString, const ErrorCategory category)
```

File: src/libsync/syncengine.cpp [hunk 10]
```diff
@@ -914,12 +914,12 @@ void SyncEngine::setNetworkLimits(int upload, int download)
     }
 }
 
-void SyncEngine::slotItemCompleted(const SyncFileItemPtr &item)
+void SyncEngine::slotItemCompleted(const SyncFileItemPtr &item, ErrorCategory category)
```
Comment (claucambra) [inline src/libsync/syncengine.cpp:917]:
```suggestion
void SyncEngine::slotItemCompleted(const SyncFileItemPtr &item, const ErrorCategory category)
```

File: src/libsync/syncengine.h [hunk 11]
```diff
@@ -163,16 +163,16 @@ public slots:
     void aboutToPropagate(OCC::SyncFileItemVector &);
 
     // after each item completed by a job (successful or not)
-    void itemCompleted(const OCC::SyncFileItemPtr &);
+    void itemCompleted(const OCC::SyncFileItemPtr &item, OCC::ErrorCategory category);
```
Comment (claucambra) [inline src/libsync/syncengine.h:166]:
```suggestion
    void itemCompleted(const OCC::SyncFileItemPtr &item, const OCC::ErrorCategory category);
```

File: src/libsync/syncengine.h [hunk 12]
```diff
@@ -163,16 +163,16 @@ public slots:
...
 
     void transmissionProgress(const OCC::ProgressInfo &progress);
 
     void itemDiscovered(const OCC::SyncFileItemPtr &);
 
     /// We've produced a new sync error of a type.
-    void syncError(const QString &message, OCC::ErrorCategory category = OCC::ErrorCategory::Normal);
+    void syncError(const QString &message, OCC::ErrorCategory category);
```
Comment (claucambra) [inline src/libsync/syncengine.h:173]:
```suggestion
    void syncError(const QString &message, const OCC::ErrorCategory category);
```

**Review by claucambra — COMMENTED** [review]
Just comments on constifying method parameters

File: src/libsync/propagatedownload.cpp [hunk 13]
```diff
@@ -461,7 +461,7 @@ void PropagateDownloadFile::start()
     SyncJournalFileRecord parentRec;
     if (!propagator()->_journal->getFileRecord(parentPath, &parentRec)) {
         qCWarning(lcPropagateDownload) << "could not get file from local DB" << parentPath;
-        done(SyncFileItem::NormalError, tr("could not get file %1 from local DB").arg(parentPath));
+        done(SyncFileItem::NormalError, tr("could not get file %1 from local DB").arg(parentPath), ErrorCategory::GenericError);
```
[the remark this answers is not in the document]
Comment (mgallien) [reply src/libsync/propagatedownload.cpp:464]: no
this will not compile

**Comment by mgallien** [comment]
latest state
image

**Review by claucambra — APPROVED** [review]

**Pull request merged**